lib/ec2/host/host_data.rb

```diff
@@ -211,7 +211,7 @@ def role_match?(condition)
           parts = role.first.split(Config.role_tag_delimiter, Config.role_max_depth)
         else
           parts = 1.upto(Config.role_max_depth).map do |i|
-            (condition["role#{i}".to_sym] || condition["usage#{i}".to_sym] || []).first
+            (condition["role#{i}".to_sym] || condition["usage#{i}".to_sym] || nil)
           end
```

Comment:
`parts` must be an array of form `[role1, role2, role3]`, but this change breaks it.